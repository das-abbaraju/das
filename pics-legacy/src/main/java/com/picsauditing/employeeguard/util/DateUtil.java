package com.picsauditing.employeeguard.util;

import com.picsauditing.employeeguard.entities.IntervalType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class DateUtil {
	private static final Logger LOG = LoggerFactory.getLogger(DateUtil.class);

	public static final Date END_OF_TIME = new Date(64060617600000l);

	public static Date explodedToDate(final int year, final int month, final int day) {
		if (year > 0 && month > 0 && day > 0) {
			Calendar calendar = Calendar.getInstance();
			calendar.set(year, month - 1, day);
			return calendar.getTime();
		}

		return null;
	}

	public static int[] dateToExploded(final Date date) {
		if (date != null) {
			int[] exploded = new int[3];
			Calendar calendar = Calendar.getInstance();
			calendar.setTime(date);

			exploded[0] = calendar.get(Calendar.YEAR);
			exploded[1] = calendar.get(Calendar.MONTH) + 1;
			exploded[2] = calendar.get(Calendar.DAY_OF_MONTH);

			return exploded;
		}

		return new int[]{};
	}

	public static boolean doesNotExpire(final Date date) {
		if (date == null) {
			return false;
		}

		if (END_OF_TIME.equals(date)) {
			return true;
		}

		Calendar dayBeforeEndOfTime = Calendar.getInstance();
		dayBeforeEndOfTime.setTime(END_OF_TIME);
		dayBeforeEndOfTime.add(Calendar.DAY_OF_YEAR, -1);

		return date.after(dayBeforeEndOfTime.getTime());
	}

	public static boolean isAboutToExpire(final Date documentExpirationDate, final int intervalPeriod, final IntervalType intervalType) {
		Calendar documentExpiration = Calendar.getInstance();
		documentExpiration.setTime(documentExpirationDate);

		switch (intervalType) {
			case DAY:
				documentExpiration.add(Calendar.DAY_OF_YEAR, -intervalPeriod);
				break;
			case WEEK:
				documentExpiration.add(Calendar.WEEK_OF_YEAR, -intervalPeriod);
				break;
			case MONTH:
				documentExpiration.add(Calendar.MONTH, -intervalPeriod);
				break;
			case YEAR:
				documentExpiration.add(Calendar.YEAR, -intervalPeriod);
				break;
			case NO_EXPIRATION:
				return false;
			case NOT_APPLICABLE:
				return false;
			default:
				throw new IllegalArgumentException();
		}

		return new Date().after(documentExpiration.getTime());
	}

  public static String getSystemTime() {
    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
    Date systemTime = new Date();

    return dateFormat.format(systemTime);
  }
}
