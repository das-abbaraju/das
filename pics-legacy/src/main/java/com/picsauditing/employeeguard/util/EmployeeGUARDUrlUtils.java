package com.picsauditing.employeeguard.util;

public final class EmployeeGUARDUrlUtils {

	/* Corporate and Operator Links */
	public static final String OPERATOR_SUMMARY = "/employee-guard/operators/dashboard";
	public static final String OPERATOR_ASSIGNMENTS = "/employee-guard/operators/assignments";
	public static final String OPERATOR_PROJECTS = "/employee-guard/operators/projects";
	public static final String OPERATOR_JOB_ROLES = "/employee-guard/operators/role";
	public static final String OPERATOR_SKILLS = "/employee-guard/operators/skill";
	public static final String OPERATOR_PROJECT_CREATE = "/employee-guard/operators/project/create";

	/* Contractor Links */
	public static final String CONTRACTOR_SUMMARY = "/employee-guard/contractor/dashboard";
	public static final String CONTRACTOR_ASSIGNMENTS = "/employee-guard/contractor/project";
	public static final String CONTRACTOR_GROUPS = "/employee-guard/contractor/employee-group";
	public static final String CONTRACTOR_SKILLS = "/employee-guard/contractor/skill";
	public static final String CONTRACTOR_EMPLOYEES = "/employee-guard/contractor/employee";
	public static final String CONTRACTOR_EMPLOYEE_IMPORT = "/contractor/employee/import-export";

	/* Employee Links */
	public static final String IMAGE_LINK = "/employee-guard/employee/contractor/%d/employee-photo/%d";
	public static final String EMPLOYEE_PROFILE = "/employee-guard/employee/profile/%d";
	public static final String EMPLOYEE_SUMMARY = "/employee-guard/employee/dashboard";
	public static final String EMPLOYEE_MY_FILES = "/employee-guard/employee/file";
	public static final String EMPLOYEE_SKILLS = "/employee-guard/employee/skills";

	public static String buildUrl(final String url, final Object... params) {
		return String.format(url, params);
	}

}
