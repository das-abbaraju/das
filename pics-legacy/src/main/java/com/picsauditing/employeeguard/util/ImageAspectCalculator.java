package com.picsauditing.employeeguard.util;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ImageAspectCalculator {

	/**
	 *
	 * Given scale=300
	 *
	 * Example 1
	 * 3000 X 1000
	 * new Width=300, new Height= (300X1000)/3000
	 *
	 * Example 2
	 * 1000 X 1000
	 * new Height=300, new Width= (300X1000)/3000
	 *
	 * Example 3
	 * 100 X 50
	 * Same values returned since both are < scale value
	 *
	 * @param scale
	 * @return
	 */
	public AspectResult calculateThumnailSize(int width, int height, int scale){
		int newWidth=0;
		int newHeight=0;

		if(width < scale && height < scale){
			return new AspectResult(width, height);
		}

		if(width ==0  || height == 0){
			return new AspectResult(width, height);
		}

		if(width > height){
			newWidth = scale;
			newHeight = (scale*height)/width;
		}
		else{
			newHeight = scale;
			newWidth =  (scale*width)/height;

		}

		return new AspectResult(newWidth, newHeight);
	}

	private class AspectResult{
		private int width;
		private int height;

		private AspectResult(int width, int height) {
			this.width = width;
			this.height = height;
		}

		public int getWidth() {
			return width;
		}

		public int getHeight() {
			return height;
		}
	}
}
