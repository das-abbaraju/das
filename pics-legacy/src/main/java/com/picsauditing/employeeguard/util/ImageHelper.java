package com.picsauditing.employeeguard.util;

import org.imgscalr.Scalr;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.*;

public class ImageHelper {

	public static final int THUMBNAIL_DEFAULT_SCALE=300;
	public static final String THUMBNAIL_DEFAULT_FORMAT="jpg";

	/**
	 *
	 * Given scale=300
	 *
	 * Example 1
	 * 3000 X 1000
	 * new Width=300, new Height= (300X1000)/3000
	 *
	 * Example 2
	 * 1000 X 1000
	 * new Height=300, new Width= (300X1000)/3000
	 *
	 * Example 3
	 * 100 X 50
	 * Same values returned since both are < scale value
	 *
	 * @param scale
	 * @return
	 */
	public static AspectResult calculateThumnailSize(int width, int height, int scale){
		int newWidth=0;
		int newHeight=0;

		if(width < scale && height < scale){
			return new AspectResult(width, height);
		}

		if(width ==0  || height == 0){
			return new AspectResult(width, height);
		}

		if(width > height){
			newWidth = scale;
			newHeight = (scale*height)/width;
		}
		else{
			newHeight = scale;
			newWidth =  (scale*width)/height;

		}

		return new AspectResult(newWidth, newHeight);
	}

	public static InputStream resizeImage(int width, int height, BufferedImage originalImage, String formatName) throws IOException {

		BufferedImage resizedImage = Scalr.resize(originalImage, Scalr.Method.BALANCED, width, height);

		ByteArrayOutputStream baos = new ByteArrayOutputStream();
		ImageIO.write(resizedImage, formatName, baos);
		InputStream is = new ByteArrayInputStream(baos.toByteArray());

		return is;
	}

	public static class AspectResult{
		private int width;
		private int height;

		private AspectResult(int width, int height) {
			this.width = width;
			this.height = height;
		}

		public int getWidth() {
			return width;
		}

		public int getHeight() {
			return height;
		}
	}
}
