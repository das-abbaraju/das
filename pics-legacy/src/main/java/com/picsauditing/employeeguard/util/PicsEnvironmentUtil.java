/**
 *
 *
 * User: aphatarphekar
 * Date: 7/17/2014
 * Time: 2:56 PM
 *
 */
package com.picsauditing.employeeguard.util;

import com.picsauditing.dao.AppPropertyDAO;
import com.picsauditing.jpa.entities.AppProperty;

import com.picsauditing.service.AppPropertyService;
import com.picsauditing.util.system.PicsEnvironment;
import org.springframework.beans.factory.annotation.Autowired;

public class PicsEnvironmentUtil {

	@Autowired
	private AppPropertyService appPropertyService;

	private PicsEnvironment picsEnvironment;

	public PicsEnvironmentUtil() {
	}

	private PicsEnvironment getPicsEnvInstance(){
		if(picsEnvironment==null)
			picsEnvironment = new PicsEnvironment(appPropertyService.getPropertyString(AppProperty.VERSION_MAJOR),
						appPropertyService.getPropertyString(AppProperty.VERSION_MINOR));

		return picsEnvironment;
	}

	public String getPicsEnvironment() {

		return getPicsEnvInstance().getEnvironment();
	}

	public boolean isLocalhostEnvironment() {
		return getPicsEnvInstance().isLocalhost();
	}
}

 