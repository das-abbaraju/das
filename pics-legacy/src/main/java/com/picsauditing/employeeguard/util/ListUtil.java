package com.picsauditing.employeeguard.util;

import java.util.Comparator;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;

public class ListUtil {
	public static <E extends Comparable<E>> List<E> removeDuplicatesAndSort(List<E> list) {
		Set<E> unique = new TreeSet<>(list);
		list.clear();
		list.addAll(unique);

		return list;
	}

	public static <E> List<E> removeDuplicatesAndSort(List<E> list, Comparator<E> comparator) {
		Set<E> unique = new TreeSet<>(comparator);
		unique.addAll(list);
		list.clear();
		list.addAll(unique);

		return list;
	}
}
