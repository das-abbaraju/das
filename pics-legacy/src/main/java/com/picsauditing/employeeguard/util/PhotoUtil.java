package com.picsauditing.employeeguard.util;

import com.picsauditing.PICS.PICSFileType;
import com.picsauditing.employeeguard.entities.DocumentType;
import com.picsauditing.employeeguard.entities.Employee;
import com.picsauditing.employeeguard.entities.Profile;
import com.picsauditing.employeeguard.entities.ProfileDocument;
import com.picsauditing.util.FileUtils;

import java.io.File;

public class PhotoUtil {
	public static final String[] VALID_PHOTO_EXTENSIONS = new String[]{"jpg", "png", "gif"};

	public static void sendPhotoToFilesDirectory(File photo, String directory, int id, String extension, String filename) throws Exception {
		FileUtils.moveFile(photo, directory, "files/" + FileUtils.thousandize(id), filename, extension, true);
	}

	public static File getPhotoForProfile(final Profile profile, final String directory) {
		ProfileDocument photoDocument = getPhotoDocumentFromProfile(profile);
		if (photoDocument != null) {
			File file = new File(directory + "/files/" + FileUtils.thousandize(profile.getId()) + photoDocument.getFileName());
			if (file.exists()) {
				return file;
			}
		}

		return null;
	}

	public static File getDefaultPhoto(final String directory) {
		File file = new File(directory + "/files/dummy.jpg");
		if (file.exists()) {
			return file;
		}

		return null;
	}

	public static ProfileDocument getPhotoDocumentFromProfile(final Profile profile) {
		for (ProfileDocument profileDocument : profile.getDocuments()) {
			if (profileDocument.getDocumentType() == DocumentType.Photo) {
				return profileDocument;
			}
		}

		return null;
	}

	public static void deleteExistingProfilePhoto(String directory, Profile profile) {
		File file = getPhotoForProfile(profile, directory);
		if (file != null && file.exists()) {
			FileUtils.deleteFile(file);
		}
	}

	public static File getPhotoForEmployee(Employee employee, int accountId, String directory) {
		int id = employee.getId();
		String filename = PICSFileType.employee_photo.filename(id) + "-" + accountId;

		File file;
		for (String extension : VALID_PHOTO_EXTENSIONS) {
			file = new File(directory + "/files/" + FileUtils.thousandize(id) + filename + "." + extension);

			if (file != null && file.exists()) {
				return file;
			}
		}

		return null;
	}

	public static boolean isValidExtension(String extension) {
		return FileUtils.checkFileExtension(extension, VALID_PHOTO_EXTENSIONS);
	}

    public static File getFile(String directory, ProfileDocument document) {
        if (document == null) {
            return null;
        }

        File file = new File(directory + "/files/" + FileUtils.thousandize(document.getProfile().getId())
                + document.getFileName());

        if (file.exists()) {
            return file;
        }

        return null;
    }
}
