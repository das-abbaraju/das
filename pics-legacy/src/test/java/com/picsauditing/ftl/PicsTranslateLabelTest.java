package com.picsauditing.ftl;

import com.picsauditing.util.Strings;
import freemarker.core.Environment;
import freemarker.template.*;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Test;

import java.io.*;
import java.util.HashMap;
import java.util.Map;

@UseReporter(DiffReporter.class)
public class PicsTranslateLabelTest {
    private static Configuration cfg;
    private Template template;
    private Map root;
    private Map parameters;
    private Map s;
    private Writer out;

    static {
        try {
            cfg = new Configuration();

            File cwd = new File(".");
            String templatePath = "src" + Strings.FILE_SEPARATOR
                    + "main" + Strings.FILE_SEPARATOR
                    + "resources" + Strings.FILE_SEPARATOR
                    + "template" + Strings.FILE_SEPARATOR
                    + "pics";

            // to run from Intellij, we're going to do some trickery. If it is maven, the absolute path will be the
            // module. If it isn't the module, we'll assume you're running from your IDE the traditioinal way, in which
            // case the root is most likely to be the project root.
            if (!cwd.getAbsolutePath().endsWith("pics-legacy" + Strings.FILE_SEPARATOR + ".")) {
                templatePath = "pics-legacy" + Strings.FILE_SEPARATOR + templatePath;
            }

            cfg.setDirectoryForTemplateLoading(new File(templatePath));
            cfg.setObjectWrapper(new DefaultObjectWrapper());
            cfg.setDefaultEncoding("UTF-8");
            cfg.setTemplateExceptionHandler(TemplateExceptionHandler.HTML_DEBUG_HANDLER);
            cfg.setIncompatibleEnhancements("2.3.19");
        } catch (Exception e) {
            // tests will fail below
        }
    }

    @Before
    public void setup() throws Exception {
        out = new StringWriter();
        root = new HashMap();
        parameters = new HashMap();
        s = new HashMap();

        parameters.put("id", "test_id");
        root.put("parameters", parameters);
        root.put("s", s);

        template = cfg.getTemplate("translate-label.ftl");
    }

    @Test
    public void test_ReturnsLabelHtmlWithTranslatedText() throws Exception {
        parameters.put("label", "Test Label");
        s.put("if", new StrutsTrueIf());
        s.put("elseif", new StrutsFalseIf());
        s.put("text", new StrutsTextHasValue());

        template.process(root, out);

        Approvals.verify(out.toString());
    }

    @Test
    public void test_ReturnsLabelHtmlWithLabelKeyAsText() throws Exception {
        parameters.put("label", "Test Label");
        s.put("if", new StrutsTrueIf());
        s.put("elseif", new StrutsFalseIf());
        s.put("text", new StrutsTextEmptyValue());

        template.process(root, out);

        Approvals.verify(out.toString());
    }

    @Test
    public void test_ReturnsLabelHtmlWithLabelKeyAsTextWhenHasKeyIsFalse() throws Exception {
        parameters.put("label", "Test Label");
        s.put("if", new StrutsFalseIf());
        s.put("elseif", new StrutsTrueIf());

        template.process(root, out);

        Approvals.verify(out.toString());
    }

    @Test
    public void test_ReturnsNothingWhenLabelKeyIsEmpty() throws Exception {
        parameters.put("label", "");
        s.put("if", new StrutsFalseIf());
        s.put("elseif", new StrutsFalseIf());

        template.process(root, out);

        Approvals.verify(out.toString());
    }

    @Test
    public void test_ReturnsLabelHtmlWithPropertyValueAsText() throws Exception {
        parameters.put("label", null);
        s.put("if", new StrutsTrueIf());
        s.put("elseif", new StrutsFalseIf());
        s.put("text", new StrutsTextEmptyValue());
        s.put("property", new StrutsPropertyValue());

        template.process(root, out);

        Approvals.verify(out.toString());
    }

    private class StrutsTrueIf implements TemplateDirectiveModel {
        @Override
        public void execute(Environment env, Map params, TemplateModel[] loopVars, TemplateDirectiveBody body) throws TemplateException, IOException {
            body.render(env.getOut());
        }
    }
    private class StrutsFalseIf implements TemplateDirectiveModel {
        @Override
        public void execute(Environment env, Map params, TemplateModel[] loopVars, TemplateDirectiveBody body) throws TemplateException, IOException {
            // do nothing
        }
    }

    private class StrutsTextHasValue implements TemplateDirectiveModel {
        @Override
        public void execute(Environment env, Map params, TemplateModel[] loopVars, TemplateDirectiveBody body) throws TemplateException, IOException {
            env.getOut().write("Translated Text");
        }
    }

    private class StrutsTextEmptyValue implements TemplateDirectiveModel {
        @Override
        public void execute(Environment env, Map params, TemplateModel[] loopVars, TemplateDirectiveBody body) throws TemplateException, IOException {
            env.getOut().write("");
        }
    }

    private class StrutsPropertyValue implements TemplateDirectiveModel {
        @Override
        public void execute(Environment env, Map params, TemplateModel[] loopVars, TemplateDirectiveBody body) throws TemplateException, IOException {
            env.getOut().write("Test Property");
        }
    }

}
