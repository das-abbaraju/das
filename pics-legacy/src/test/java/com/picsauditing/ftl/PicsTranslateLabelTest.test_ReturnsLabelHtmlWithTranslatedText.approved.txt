
    <label for="test_id">
            Translated Text
    </label>
